packages/preview-middleware/src/base/config.ts

```diff
@@ -0,0 +1,396 @@
+
+export interface CustomConnector {
+    applyConnector: string;
+    writeConnector: string;
+    custom: boolean;
+}
+
+export interface FlexConnector {
+    connector: string;
+    layers: string[];
+    url?: string;
+}
+
+/**
+ * Internal structure used to fill the sandbox.html template
+ */
+export interface TemplateConfig {
+    basePath: string;
+    apps: Record<
+        string,
+        {
+            title: string;
+            description: string;
+            additionalInformation: string;
+            applicationType: 'URL';
+            url: string;
+            applicationDependencies?: { manifest: boolean };
+        }
+    >;
+    ui5: {
+        libs: string;
+        theme: string;
+        flex: (CustomConnector | FlexConnector)[];
+        bootstrapOptions: string;
+        resources: Record<string, string>;
+    };
+    init?: string;
+    flex?: {
+        [key: string]: unknown;
+        layer: UI5FlexLayer;
+        developerMode: boolean;
+        pluginScript?: string;
+    };
+    locateReuseLibsScript?: boolean;
+}
+
+/**
+ * Static settings
+ */
+export const PREVIEW_URL = {
+    client: {
+        url: '/preview/client',
+        local: join(__dirname, '../../dist/client'),
+        ns: 'open.ux.preview.client'
+    },
+    api: '/preview/api'
+} as const;
+
+/**
+ * Default theme
+ */
+export const DEFAULT_THEME = 'sap_horizon';
+
+/**
+ * Default path for mounting the local FLP.
+ */
+export const DEFAULT_PATH = '/test/flp.html';
+
+/**
+ * Default intent
+ */
+export const DEFAULT_INTENT = {
+    object: 'app',
+    action: 'preview'
+} as Readonly<Intent>;
+
+/**
+ * SAPUI5 delivered namespaces from https://ui5.sap.com/#/api/sap
+ */
+const UI5_LIBS = [
+    'sap.apf',
+    'sap.base',
+    'sap.chart',
+    'sap.collaboration',
+    'sap.f',
+    'sap.fe',
+    'sap.fileviewer',
+    'sap.gantt',
+    'sap.landvisz',
+    'sap.m',
+    'sap.ndc',
+    'sap.ovp',
+    'sap.rules',
+    'sap.suite',
+    'sap.tnt',
+    'sap.ui',
+    'sap.uiext',
+    'sap.ushell',
+    'sap.uxap',
+    'sap.viz',
+    'sap.webanalytics',
+    'sap.zen'
+] as const;
+
+/**
+ * Gets the UI5 libs dependencies from manifest.json.
+ *
+ * @param manifest application manifest
+ * @returns UI5 libs that should preloaded
+ */
+function getUI5Libs(manifest: Manifest): string {
+    const libs = manifest['sap.ui5']?.dependencies?.libs ?? {};
+    // add libs that should always be preloaded
+    libs['sap.m'] = {};
+    libs['sap.ui.core'] = {};
+    libs['sap.ushell'] = {};
+
+    return Object.keys(libs)
+        .filter((key) => {
+            return UI5_LIBS.some((substring) => {
+                return key === substring || key.startsWith(substring + '.');
+            });
+        })
+        .join(',');
+}
+
+/**
+ * Default configuration for the FLP.
+ *
+ * @param config partial configuration
+ * @returns a full configuration with default values
+ */
+export function getFlpConfigWithDefaults(config: Partial<FlpConfig> = {}) {
```

Comment:
It doesn't return an FLP config, necessarily, right? If config.path is different, we can also address other preview options, hopefully.
SHouldn't the name of the function be more general, then? Like getPrevieConfig?

Reply:
The preview config contains a four main objects: adp (for adaptation projects), rta (for configuring editors), test (for configuring test suites) and lastly flp for configuring the FLP preview. This function returns the config for the FLP preview with defaults.

Reply:
OK, I checked the README now